Pop-Grupa-A-Backend/app/main/repositories/user_repository.py

```diff
@@ -25,6 +41,35 @@ def add_user(user):
         }
         return response_object, 409
 
+def check_user(user):
+    db_user = ComputationAccount.query.filter_by(username=user['username']).first()
+    if not db_user:
+        response_object = {
+            'status': 'fail',
+            'messege': 'User login failure',
+        }
+        return  response_object, 403
+    if not flask_bcrypt.check_password_hash(db_user.password, user['password']):
+        response_object = {
+            'status': 'fail',
+            'messege': 'User login failure',
+        }
+        return response_object, 403
+    else:
+        sid = str(uuid.uuid4())
+        response_object = {
+            'status': 'success',
+            'messege': 'User successfully logged',
+        }
+        session['sid'] = sid
+        session['username'] = user['username']
+        new_session = Session(
+            sid=sid,
+            exp=datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+        )
+        save_changes(new_session)
+        return response_object, 200
```

Comment:
Może nie widzę ale gdzieś ten sid jest zwracany żeby można go było użyć przy kolejnych zapytaniach?

Reply:
Jeśli to dobrze zrobili to sid będzie w cookiesach, więc dzięki headerowi który dodali 'Access-Control-Allow-Credentials' powinien chyba automatycznie być przesyłany podczas requestow.